hudi-common/src/main/java/org/apache/hudi/expression/ArrayData.java

```diff
@@ -16,22 +16,25 @@
  * limitations under the License.
  */
 
-package org.apache.hudi.hive.expression;
```

Comment:
Any specific reason why we are renaming packages ?

Reply:
These expressions were first introduced in pr https://github.com/apache/hudi/pull/6725 to allow filter partitions when sync to HMS, so they were in `hudi-hive-sync` before, and now these expressions actually can be reused, so I moved them to `hudi-common` package